Orange/widgets/unsupervised/owdistancetransformation.py

```diff
@@ -0,0 +1,64 @@
+import numpy as np
+
+from Orange.util import scale
+from Orange.misc import DistMatrix
+from Orange.widgets import widget, gui, settings
+
+
+class OWDistanceTransformation(widget.OWWidget):
+    name = "Distance Transformation"
+    description = "Transform distances according to selected criteria."
+    icon = "icons/DistancesTransformation.svg"
+
+    inputs = [("Distances", DistMatrix, "set_data")]
+    outputs = [("Distances", DistMatrix)]
+
+    want_main_area = False
+
+    normalization_method = settings.Setting(0)
+    inversion_method = settings.Setting(0)
+
+    normalization_options = (
+        ("No normalization", lambda x: x),
+        ("To interval [0, 1]", lambda x: scale(x, min=0, max=1)),
+        ("Sigmoid function: 1/(1+exp(-x))", lambda x: 1/(1+np.exp(-x))),
+    )
+
+    inversion_options = (
+        ("No inversion", lambda x: x),
+        ("-X", lambda x: -x),
+        ("1 - X", lambda x: 1-x),
+        ("Max - X", lambda x: np.max(x) - x),
+        ("1/X", lambda x: 1/x),
+    )
+
+    def __init__(self):
+        super().__init__()
+
+        self.data = None
+
+        box = gui.widgetBox(self.controlArea, "Normalization")
+        gui.radioButtons(box, self, "normalization_method",
+                         btnLabels=[x[0] for x in self.normalization_options],
+                         callback=self.commit)
+
+        box = gui.widgetBox(self.controlArea, "Inversion")
+        gui.radioButtons(box, self, "inversion_method",
+                         btnLabels=[x[0] for x in self.inversion_options],
+                         callback=self.commit)
+
+    def set_data(self, data):
+        self.data = data
+        self.commit()
+
+    def commit(self):
```

Comment:
Is it ok that all these are applied without a _Commit_ button? I mean, I don't mind, but it's a bit non-standard.